mlflow/tracking/fluent.py

```diff
@@ -1635,18 +1635,44 @@ def _get_or_start_run():
 
 def _get_experiment_id_from_env():
     experiment_name = env.get_env(_EXPERIMENT_NAME_ENV_VAR)
+    experiment_id = env.get_env(_EXPERIMENT_ID_ENV_VAR)
     if experiment_name is not None:
         exp = MlflowClient().get_experiment_by_name(experiment_name)
-        return exp.experiment_id if exp else None
-    return env.get_env(_EXPERIMENT_ID_ENV_VAR)
+        if exp:
+            if experiment_id and experiment_id != exp.experiment_id:
+                raise MlflowException(
+                    message=f"The provided environment variable {_EXPERIMENT_ID_ENV_VAR} "
+                    f"`{experiment_id}` does not match the experiment id "
+                    f"`{exp.experiment_id}` for experiment name `{experiment_name}`",
+                    error_code=INVALID_PARAMETER_VALUE,
+                )
+            else:
+                return exp.experiment_id
+        else:
+            raise MlflowException(
+                message=f"The provided environment variable {_EXPERIMENT_NAME_ENV_VAR} "
+                f"`{experiment_name}` does not exist. Create an experiment with this name "
+                f"by using `mlflow.create_experiment(name='{experiment_name}')`",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
```

Comment:
Can we create the experiment in this case instead of failing? E.g. https://github.com/mlflow/mlflow/blob/b54457db9c992e27a4a403311429e26024aff717/mlflow/tracking/fluent.py#L123

Reply:
definitely!